### PR TITLE
lcampbell/genome prepare

### PR DESCRIPTION
**Pull request by ens-LCampbell**
Added some slight changes to amend_genome_data and added option for collecting output files into a single user specified output directory. 

Note, all changes were done on top of @MatBarba most recent large scale code cleanup/refactoring commits to main branch.

### CODE REVIEW

File: src/python/ensembl/io/genomio/metadata/update_genome_metadata.py [hunk 1]
```diff
@@ -183,7 +180,15 @@ class InputSchema(argschema.ArgSchema):
...
-    brc4_mode = argschema.fields.Int(required=False, metadata={"description": "Activate BRC4 mode (default)"})
+    brc4_mode = argschema.fields.Int(
+        required=False,
+        metadata={"description": "Activate BRC4 mode (default)"}
+    )
+    output_json = argschema.fields.OutputFile(
+        required=False,
+        dump_default="amend_genome_meta.json",
+        metadata={"description": "Default json file to capture json metadata"},
+    )
```
Comment (MatBarba) [inline src/python/ensembl/io/genomio/metadata/update_genome_metadata.py:191]: Minor: can you run black for this file?

File: pipelines/nextflow/subworkflows/genome_prepare/genome_prepare.nf [hunk 2]
```diff
@@ -61,39 +64,47 @@ workflow GENOME_PREPARE {
...
+        //             .groupTuple()
+
+        // // Collect in manifest, checks and stats
+        // collect_dir = COLLECT_FILES(prepared_files)
+        // manifest_dired = MANIFEST(collect_dir, accession)
+        // manifest_checked = CHECK_INTEGRITY(manifest_dired, params.brc4_mode)
+        // manifest_stated = MANIFEST_STATS(manifest_checked, 'datasets')
+
+        // // Publish the data
+        // PUBLISH_DIR(manifest_stated, output_dir, accession)
```
Comment (MatBarba) [inline pipelines/nextflow/subworkflows/genome_prepare/genome_prepare.nf:109]: I assume the pipeline is not finished at this point?

**Review by MatBarba — APPROVED** [review]
That looks fine to me

File: src/python/ensembl/io/genomio/metadata/update_genome_metadata.py (hunk 1 above)
Comment (ens-LCampbell) [reply src/python/ensembl/io/genomio/metadata/update_genome_metadata.py]: Blacked ! Sorry my bad..... I had formatted it to be inline with other option params

File: pipelines/nextflow/subworkflows/genome_prepare/genome_prepare.nf (hunk 2 above)
Comment (MatBarba) [reply pipelines/nextflow/subworkflows/genome_prepare/genome_prepare.nf:109]: I think that's not going to work unless you keep the accession associated with each file along the whole pipeline (that's the only way to make sure we keep them in sync). This is really tricky and I'll need to focus on this once I'm done with other things...

That said, you can push the PR and we can work on that next part afterwards.

**Review by MatBarba — APPROVED** [review]

**Pull request merged**